src/bespokelabs/curator/request_processor/base_online_request_processor.py

```diff
@@ -134,13 +137,22 @@ def __init__(
         self.prompt_formatter: Optional[PromptFormatter] = None
         self.max_requests_per_minute: Optional[int] = max_requests_per_minute
         self.max_tokens_per_minute: Optional[int] = max_tokens_per_minute
-        self.DEFAULT_MAX_REQUESTS_PER_MINUTE = DEFAULT_REQUESTS_PER_MINUTE
-        self.DEFAULT_MAX_TOKENS_PER_MINUTE = DEFAULT_TOKENS_PER_MINUTE
+        if max_retries is None:
+            self.max_retries = DEFAULT_MAX_RETRIES
+        else:
+            self.max_retries = max_retries
 
     def get_rate_limit(self, name, header_value):
         """Uses manual values if set, otherwise uses headers if available, and if not available uses defaults."""
-        manual_value = getattr(self, name)
-        default_value = getattr(self, f"DEFAULT_{name.upper()}")
+        if name == "max_requests_per_minute":
+            manual_value = self.max_requests_per_minute
+            default_value = DEFAULT_MAX_REQUESTS_PER_MINUTE
+        elif name == "max_tokens_per_minute":
+            manual_value = self.max_tokens_per_minute
+            default_value = DEFAULT_MAX_TOKENS_PER_MINUTE
+        else:
+            raise ValueError(f"Invalid rate limit name: {name}")
```

Comment:
I think this is the right way to do this: you use properties or separate methods.

```
__init__(...
   max_requests_per_minute: Optional[int] = None
  ...
):
  self._max_requests_per_minute = max_requests_per_minute

@property
def max_requests_per_minute(self):
  if self._max_requests_per_minute:
    return self._max_requests_per_minute
  else:
     return DEFAULT_MAX_REQUESTS_PER_MINUTE
```

And then perhaps you can get rid of this `get_rate_limit` method and just access it: `self.max_requests_per_minute`.

Or if you need to process baed on header, you do this:

```
def max_requests_per_minute(self, header):
  header_rpm = ..
  ...
```